lib/warehouse_messenger/dao/container_dao.pm

```diff
@@ -0,0 +1,207 @@
+package warehouse_messenger::dao::container_dao;
+
+use Moose;
+use Readonly;
+use Carp;
+
+use warehouse_messenger::dao::containertypes_dao;
+use warehouse_messenger::dao::sample_dao;
+use warehouse_messenger::dao::study_dao;
+
+with 'warehouse_messenger::dao::base_dao';
+
+# In the ATTRIBUTES hash: an element's key is the attribute name
+# and the element's value is the XPATH to get the attribute's value
+
+## no critic(ValuesAndExpressions::RequireInterpolationOfMetachars)
+Readonly::Hash  my %ATTRIBUTES  => {
+  id_flgen_plate_lims => q{/con:container/@limsid},
+  plate_barcode_lims  => q{/con:container/name},
+};
+
+Readonly::Scalar my $TYPE_URI_PATH => q{/con:container/type/@uri};
+Readonly::Scalar my $ARTIFACT_PATH      => q{/con:container/placement/@limsid};
+Readonly::Scalar my $ARTIFACT_LOCATION  => q{/art:artifact/location/value};
+Readonly::Scalar my $SAMPLE_LIMSID_PATH => q{/art:artifact/sample/@limsid};
+## use critic
+
+our $VERSION = '0.0';
+
+has '+resource_type' => (
+  default     => 'containers',
+);
+
+has '+attributes' => (
+  default     => sub { return \%ATTRIBUTES; },
+);
+
+has 'type' => (
+  traits     => [ 'DoNotSerialize' ],
+  is         => 'ro',
+  isa        => 'warehouse_messenger::dao::containertypes_dao',
+  lazy_build => 1,
+);
+
+sub _build_type {
+  my $self = shift;
+  my $type_uri = $self->findvalue($TYPE_URI_PATH);
+  my $type_limsid;
+
+  if ($type_uri =~ /containertypes\/(\d+)$/sxm) {
+    $type_limsid = $1;
+  } else {
+    croak 'Containertype does not have an id';
+  }
+
+  return warehouse_messenger::dao::containertypes_dao->new(lims_id => $type_limsid);
+}
+
+has 'plate_size' => (
+  is => 'ro',
+  isa => 'Int',
+  lazy_build => 1,
+);
+
+sub _build_plate_size {
+  my $self = shift;
+  return $self->type->plate_size;
+}
+
+sub flgen_well_position {
+  my ($self, $loc, $nb_rows, $nb_cols) = @_;
+
+  if (!$loc) {
+    croak 'Well address should be given';
+  }
+
+  my ($letter, $number) = $loc =~ /\A(\w):(\d+)/xms;
+  if (!$letter || !$number) {
```

Comment:
How about this regexp?: ^([A-P]+):(\d{1,2})$
With yours this $loc: 'A:1TEST' is also passing.